### ISSUE TITLE
FeatureNames doesn't support slicing

### ISSUE THREAD

**Issue by kmike** [TeamHG-Memex/eli5#97]
`ivec.get_feature_names()[:100]` doesn't work for InvertableHashingVectorizer because FeatureName doesn't handle slice objects.

See also: http://stackoverflow.com/questions/13855288/turn-slice-into-range-in-python; there is also a nice comment

> Does it help that in Python 3 you can slice a range object to get a new range object?

-- given that a feature is not super-important we could decide to support it only in Python 3 :)

**Comment by lopuhin** [comment]
Nice catch!
I tried a negative slice in python 2, and it turns it into a non-negative slice, so it should be easy to support it here too.
That slice operation should return a new FeatureNames object, right?

**Comment by kmike** [comment]
> That slice operation should return a new FeatureNames object, right?

A good question; I think that's fine to return a list if returning FeatureUnion is harder to implement.

**Comment by lopuhin** [comment]
Hm, I didn't think about it at first (was going to make it return FeatureUnion), but in fact it will make implementation more complex - for example, we still might have missing entries that are handled by ``unkn_template``, but now the index must be shifted.
So I agree, let's go with returning a list (or maybe a numpy slice if the underlying storage is a numpy array)
